Client crash (NPE) in ClientRequestManager Hashtable.put during document-manager load (macOS + Windows)

## Summary

A `NullPointerException` originating in `ClientRequestManager` (a `Hashtable.put`
at `ClientRequestManager.java:3528`) crashes a background client task. Seen on
**two independent reports** (macOS and Windows), same line — appears to be a null
key or value being put into a `Hashtable` while loading workspace/document
information after sign-in.

Reported via two auto-submitted client crash reports.

## Environment

- VCell `Rel_Version_8.0.0_build_03`
- Report 1: macOS 15.7.3 (aarch64), Java 17.0.14
- Report 2: Windows 11 (amd64), Java 17.0.7

## Observed

```
java.lang.RuntimeException: ...
Caused by: java.lang.NullPointerException
    at java.base/java.util.Hashtable.put(Hashtable.java:476)
    at cbit.vcell.client.ClientRequestManager$39.run(ClientRequestManager.java:3528)
    at cbit.vcell.client.task.ClientTaskDispatcher.runTask(ClientTaskDispatcher.java:677)
    at cbit.vcell.client.task.ClientTaskDispatcher$1.construct(ClientTaskDispatcher.java:380)
    at swingthreads.SwingWorker$2.run(SwingWorker.java:109)
```

Surrounding client log on both reports shows document-manager initialization
around the failure (`ClientDocumentManager.initAllDatabaseInfos()`,
`ClientDocumentManager.getPreferences()`), and on the Windows report repeated
`>> polling failure <<` lines shortly before.

## Expected

The client task should not crash on a null map key/value; the `Hashtable.put` at
`ClientRequestManager.java:3528` should tolerate/guard the missing value (or fail
with an actionable message) rather than throwing an NPE on the worker thread.

## Notes

Because `Hashtable` rejects null keys **and** null values, the fix likely needs
to identify which is null at that call site and handle the missing entry. The
`polling failure` lines suggest a possible correlation with a degraded server
connection at load time.


---
_Source: VCell support email (Message-ID `c85cc1c579b349ed8a57be42a4c7b6b2@MN6PR14MB7148.namprd14.prod.outlook.com and 733f584cfd0a47a487aaae3990c99d88@MN6PR14MB7148.namprd14.prod.outlook.com`). Reporter details withheld._

--- a/vcell-client/src/main/java/cbit/vcell/client/ClientRequestManager.java
+++ b/vcell-client/src/main/java/cbit/vcell/client/ClientRequestManager.java
@@ -3161,7 +3161,9 @@ private BioModel createDefaultBioModelDocument(BngUnitSystem bngUnitSystem) thro
 			ExternalDocInfo externalDocInfo = (ExternalDocInfo) documentInfo;
 
 			File file = externalDocInfo.getFile();
-			if (file != null && !file.getName().isEmpty() && file.getName().endsWith("bngl")) {
+			// lower-cased to match the sibling .sedx/.omex/.ssld checks: a file named
+			// Model.BNGL used to skip this pass, then fail as an unrecognized format
+			if (file != null && !file.getName().isEmpty() && file.getName().toLowerCase().endsWith(".bngl")) {
 
 				BngUnitSystem bngUnitSystem = new BngUnitSystem(BngUnitOrigin.DEFAULT);
 				String fileText;
@@ -3496,13 +3498,32 @@ private BioModel createDefaultBioModelDocument(BngUnitSystem bngUnitSystem) thro
 							}
 						}
 					}
-					if (doc == null && docs == null) {
-						File f = externalDocInfo.getFile();
-						if (f != null) {
-							throw new RuntimeException("Unable to determine type of file " + f.getCanonicalPath());
-						}
-						throw new ProgrammingException();
+				}
+				// Nothing was produced: the load or import failed without saying so. Until this
+				// was caught here it surfaced as an NPE out of Hashtable.put("doc", null) just
+				// below, which is what users actually saw (issue #1748).
+				//
+				// The previous version of this check was dead code: it tested "docs == null",
+				// but docs is initialized to an empty list and is never null. It also sat
+				// inside the ExternalDocInfo branch, so it could never have covered the other
+				// document types either way.
+				if (doc == null && docs.isEmpty()) {
+					File f = (documentInfo instanceof ExternalDocInfo)
+							? ((ExternalDocInfo) documentInfo).getFile()
+							: null;
+					if (f != null) {
+						throw new RuntimeException("Unable to determine the type of file '" + f.getCanonicalPath()
+								+ "'. It is not a recognized VCML, SBML, SED-ML/OMEX, BNGL or SSLD document.");
 					}
+					// No file: the content arrived as a string, e.g. SBML fetched from
+					// BioModels.net. An empty or non-XML payload means that retrieval failed.
+					String name = (documentInfo instanceof ExternalDocInfo)
+							? ((ExternalDocInfo) documentInfo).getDefaultName()
+							: null;
+					throw new RuntimeException("Unable to read the document"
+							+ (name == null ? "" : " '" + name + "'")
+							+ ": the content is empty or not in a recognized format."
+							+ " If it was retrieved from a remote database, that request may have failed.");
 				}
 				// create biopax objects using annotation
 				if (doc instanceof BioModel) {

